### PR TITLE
docs: add route and upstream mtls

### LINKED ISSUE #4327: Extend the admin mTLS doc to cover all the mTLS concepts
Currently we don't have a united place to hold all the mTLS info.
Let's write one from the users' aspect.
CC @lingsamuel

### PR DESCRIPTION
**Pull request by lingsamuel**
Signed-off-by: Ling Samuel <lingsamuelgrace@gmail.com>

### What this PR does / why we need it:
Fixes #4327

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [x] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: docs/en/latest/mtls.md [hunk 1]
```diff
@@ -59,3 +61,115 @@ Please replace the following certificate paths and domain name with your real on
...
 curl --cacert /data/certs/mtls_ca.crt --key /data/certs/mtls_client.key --cert /data/certs/mtls_client.crt  https://admin.apisix.dev:9180/apisix/admin/routes -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1'
 ```
+
+## Protect Route
+
+### Why use it
+
+Using mTLS is a way to verify clients cryptographically. It is useful and important in cases where you want to have encrypted and secure traffic in both directions.
+
+### How to config
```
Comment (spacewander) [inline docs/en/latest/mtls.md:71]: Use configure but not config in the doc

File: docs/en/latest/mtls.md [hunk 2]
```diff
@@ -59,3 +61,115 @@ Please replace the following certificate paths and domain name with your real on
...
+    "key": key,
+    "snis": [sni],
+}
+if len(sys.argv) >= 5:
+    print("Setting mTLS")
+    reqParam["client"] = {}
+    with open(sys.argv[4]) as f:
+        clientCert = f.read()
+        reqParam["client"]["ca"] = clientCert
+    reqParam["client"]["depth"] = int(sys.argv[5])
```
Comment (spacewander) [inline docs/en/latest/mtls.md:106]: Will there be an error when len(sys.argv) == 5?

File: docs/en/latest/mtls.md [hunk 3]
```diff
@@ -59,3 +61,115 @@ Please replace the following certificate paths and domain name with your real on
...
+curl --resolve 'mtls.test.com:<APISIX_HTTPS_PORT>:<APISIX_URL>' "https://<APISIX_URL>:<APISIX_HTTPS_PORT>/hello" -k --cert ./client.pem --key ./client.key
+```
+
+Please make sure that the SNI fits the certificate domain.
+
+## mTLS Between APISIX and Upstream
+
+### Why use it
+
+Sometimes the upstream enabled mTLS. In this situation, the APISIX acts as the client, it needs to provide client certificate to communicate with upstream.
```
Comment (spacewander) [inline docs/en/latest/mtls.md:129]: enabled => enables

It happens now.

File: docs/en/latest/mtls.md [hunk 4]
```diff
@@ -59,3 +61,115 @@ Please replace the following certificate paths and domain name with your real on
...
+    "snis": [sni],
+}
+if len(sys.argv) >= 5:
+    print("Setting mTLS")
+    reqParam["client"] = {}
+    with open(sys.argv[4]) as f:
+        clientCert = f.read()
+        reqParam["client"]["ca"] = clientCert
+    reqParam["client"]["depth"] = int(sys.argv[5])
+resp = requests.put("http://127.0.0.1:9180/apisix/admin/ssl/1", json=reqParam, headers={
```
Comment (spacewander) [inline docs/en/latest/mtls.md:107]: We use 9080 by default.

File: docs/en/latest/mtls.md [hunk 5]
```diff
@@ -59,3 +61,115 @@ Please replace the following certificate paths and domain name with your real on
...
+
+### Why use it
+
+Sometimes the upstream enabled mTLS. In this situation, the APISIX acts as the client, it needs to provide client certificate to communicate with upstream.
+
+### How to config
+
+When configuring `upstreams`, we could use parameter `tls.client_cert` and `tls.client_key` to config the client certificate APISIX used to communicate with upstreams.
+
+This feature requires APISIX to run on [APISIX-OpenResty](../how-to-build.md#6-build-openresty-for-apisix).
```
Comment (spacewander) [inline docs/en/latest/mtls.md:135]: Bad link

**Review by spacewander — CHANGES_REQUESTED** [review]

File: docs/en/latest/mtls.md (hunk 3 above)
Comment (lingsamuel) [reply docs/en/latest/mtls.md:129]: *Existed upstream* uses mTLS, so we need to configure mTLS in APISIX. The upstream mTLS already configured before we configure APISIX.

File: docs/en/latest/mtls.md (hunk 4 above)
Comment (lingsamuel) [reply docs/en/latest/mtls.md:107]: YAML above uses 9180 https://github.com/apache/apisix/pull/4348/files#diff-245d1b166ec738da077d9e8fa92c62adf28099b762a9806bb63132a6c20b0efaR39

File: docs/en/latest/mtls.md (hunk 3 above)
Comment (spacewander) [reply docs/en/latest/mtls.md:129]: The past tense looks like the upstream used to enable upstream, but it doesn't now.

File: docs/en/latest/mtls.md (hunk 4 above)
Comment (spacewander) [reply docs/en/latest/mtls.md:107]: This section is not relative to admin mTLS.
People won't read through the whole doc. They just jump into a section.

File: docs/en/latest/mtls.md [hunk 6]
```diff
@@ -97,7 +97,7 @@ reqParam = {
     "key": key,
     "snis": [sni],
 }
-if len(sys.argv) >= 5:
+if len(sys.argv) >= 6:
```
Comment (spacewander) [inline docs/en/latest/mtls.md:100]: The `depth` is optional to the mTLS feature. We should not require it.

File: docs/en/latest/mtls.md [hunk 7]
```diff
@@ -130,9 +130,9 @@ Sometimes the upstream enabled mTLS. In this situation, the APISIX acts as the c
 
 ### How to config
```
Comment (spacewander) [inline docs/en/latest/mtls.md:131]: Need to fix it

File: docs/en/latest/mtls.md [hunk 8]
```diff
@@ -59,3 +61,115 @@ Please replace the following certificate paths and domain name with your real on
...
+
+### Why use it
+
+Sometimes the upstream enabled mTLS. In this situation, the APISIX acts as the client, it needs to provide client certificate to communicate with upstream.
+
+### How to config
+
+When configuring `upstreams`, we could use parameter `tls.client_cert` and `tls.client_key` to configure the client certificate APISIX used to communicate with upstreams.
+
+This feature requires APISIX to run on [APISIX-OpenResty](./how-to-build.md#6-build-openresty-for-apisix).
```
Comment (Yiyiyimu) [inline docs/en/latest/mtls.md:136]:
```suggestion
This feature requires APISIX to run with [APISIX-OpenResty](./how-to-build.md#6-build-openresty-for-apisix).
```

File: docs/en/latest/mtls.md (hunk 3 above)
Comment (lingsamuel) [reply docs/en/latest/mtls.md:130]: What about: `upstream requires mTLS`?

**Review by Yiyiyimu — COMMENTED** [review]
Do we need to also change Chinese docs in this PR

**Review by spacewander — APPROVED** [review]

File: docs/en/latest/mtls.md (hunk 8 above)
Comment (lingsamuel) [reply docs/en/latest/mtls.md:149]: This line copied from https://github.com/apache/apisix/blob/master/docs/en/latest/admin-api.md#request-body-parameters-3

File: docs/zh/latest/mtls.md [hunk 9]
```diff
@@ -50,10 +54,136 @@ apisix reload
...
+
+## 保护路由
+
+### 为什么使用
+
+双向认证是一种密码学安全的验证客户端身份的手段。当你需要加密并保护流量的双向安全时很有用。
+
+### 如何配置
+
+在配置 `ssl` 资源时，同时需要配置 `client.ca` 和 `client.depth` 参数，分别代表为客户端证书签名的 CA 列表，和证书链的最大深度。可参考：[SSL API 文档](./admin-api.md#ssl)。
```
Comment (spacewander) [inline docs/zh/latest/mtls.md:86]: `可参考：[SSL API 文档](./admin-api.md#ssl)。`
The English version doesn't have it

File: docs/zh/latest/mtls.md [hunk 10]
```diff
@@ -50,10 +54,136 @@ apisix reload
...
+
+### 为什么使用
+
+有时候上游的服务启用了双向认证。在这种情况下，APISIX 作为上游服务的客户端，需要提供客户端证书来正常与其进行通信。
+
+### 如何配置
+
+在配置 upstream 资源时，可以使用参数 `tls.client_cert` 和 `tls.client_key` 来配置 APISIX 用于与上游进行通讯时使用的证书。可参考 [Upstream API 文档](./admin-api.md#upstream)。
+
+该功能需要 APISIX 运行在 [APISIX-OpenResty](./how-to-build.md#6-build-openresty-for-apisix) 上。
```
Comment (spacewander) [inline docs/zh/latest/mtls.md:149]: The link should be corrected.

File: docs/zh/latest/mtls.md [hunk 11]
```diff
@@ -50,10 +54,136 @@ apisix reload
...
+
+## APISIX 与上游间的双向认证
+
+### 为什么使用
+
+有时候上游的服务启用了双向认证。在这种情况下，APISIX 作为上游服务的客户端，需要提供客户端证书来正常与其进行通信。
+
+### 如何配置
+
+在配置 upstream 资源时，可以使用参数 `tls.client_cert` 和 `tls.client_key` 来配置 APISIX 用于与上游进行通讯时使用的证书。可参考 [Upstream API 文档](./admin-api.md#upstream)。
```
Comment (spacewander) [inline docs/zh/latest/mtls.md:147]: The English version doesn't have it

File: docs/zh/latest/mtls.md [hunk 12]
```diff
@@ -21,15 +21,19 @@ title: Admin API TLS 双向认证
...
+
 ### 为什么使用
 
-双向认证可以更好的防止未经授权访问 APISIX ，客户端将向服务器提供其证书，服务器将检查证书是否由提供的 CA 签名并决定是否响应请求。
+双向认证提供了一种更好的方法来阻止未经授权的对 APISIX Admin API 的访问。
+
+客户端需要向服务器提供证书，服务器将检查该客户端证书是否由受信的 CA 签名，并决定是否响应其请求。
 
-### 如何开启
+### 如何配置
```
[the remark this answers is not in the document]
Comment (lingsamuel) [reply docs/zh/latest/mtls.md:32]: It's to ensure consistency with other subsections.

[the remark this answers is not in the document]
Comment (spacewander) [reply docs/zh/latest/mtls.md:32]: Should fix the English doc first.

File: docs/zh/latest/mtls.md [hunk 13]
```diff
@@ -66,7 +66,7 @@ curl --cacert /data/certs/mtls_ca.crt --key /data/certs/mtls_client.key --cert /
 
 ### 如何配置
 
-你需要构建 [APISIX-Openresty](./how-to-build.md#6-build-openresty-for-apisix)，并且需要在配置文件中设定 `etcd.tls` 来使 ETCD 的双向认证功能正常工作。
+你需要构建 [APISIX-Openresty](./how-to-build.md#为-APISIX-构建-OpenResty)，并且需要在配置文件中设定 `etcd.tls` 来使 ETCD 的双向认证功能正常工作。
```
Comment (spacewander) [inline docs/zh/latest/mtls.md:69]: Should be `6-为-apisix-构建-openresty`

**Review by spacewander — APPROVED** [review]

**Review by Yiyiyimu — APPROVED** [review]

**Pull request merged**